c7n/resources/cloudfront.py

```diff
@@ -149,6 +149,56 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('logging-enabled')
+class IsLoggingEnabled(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: logging-enabled
+                resource: distribution
+                filters:
+                  - type: logging-enabled
+                    value: false
+   """
+
+    schema = type_schema(
+        'logging-enabled',
+        value={'type': 'boolean'})
+
+    permissions = ('cloudfront:GetDistributionConfig',)
+
+    def process(self, resources, event=None):
+        results = []
+
+        distribution_client = local_session(self.manager.session_factory).client(
+            'cloudfront', region_name=self.manager.config.region)
+
+        # Default to filtering for distributions with logging enabled
+        value = self.data.get('value', True)
+
+        for r in resources:
+            try:
+                config_data = distribution_client.get_distribution_config(Id=r['Id']) \
```

Comment:
let's switch this to a `distribution-config` filter, not a separate filter for each attribute in the config.

Reply:
From checking https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudfront.html#CloudFront.Client.get_distribution_config and https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudfront.html#CloudFront.Client.list_distributions, the only attribute that can't already be filtered in DistributionConfig using a value filter is Logging. Do we want to duplicate all of these values in a new filter? The other option would be to change the resource itself to use the result of get_distribution_config instead of DistributionList.Items.

Reply:
you should be comparing actual results of the methods to verify not the docs.. the docs lie due to shared shape usage. see my reply here https://github.com/boto/boto3/issues/1855#issuecomment-460059883

Reply:
Oh thanks, I didn't realize that. I verified the results of a few different policies, and 
```policies:
  - name: logging-enabled
    resource: distribution
    filters:
      - type: value
        key: "Logging.Enabled"
        value: True
```

indeed does not work as expected, as it will always match no distribution, whereas 
```policies:
  - name: logging-enabled
    resource: distribution
    filters:
      - type: value
        key: "Logging.Enabled"
        value: Null
```

will match every distribution. Also, filtering on other attributes, like 

```policies:
  - name: logging-enabled
    resource: distribution
    filters:
      - type: value
        key: "Comment"
        value: "test
```

works as expected.

Reply:
wrt to a distribution-config filter, you have to compare *all* attributes returned by get distribution config to those returned in list distributions across all permutations of distributions.. there is likely something besides logging thats differential,  this isn't something you can do in a policy to check. we don't want to add one off vocabulary for a resource when something more generic is an option.

Reply:
hey @Lucas-Irvine, as discussed lets try and do this by using ValueFilter as a base class so that there is just one filter type for all things distribution-config. I agree with @kapilt that this is the right approach and helps to not bloat that file, it also helps future proof this kind of filter in case the response data is ever changed in the future.

Reply:
Yep thanks for showing me this approach. It makes a lot of sense to do it this way.